### PR TITLE
make channel message html human readable

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds two-step processing to the html previews for channel messages and replies.  First, all inline attachments are replaced with the string `[attachment:name]`.
Second, remaining html is stripped out, leaving only plaintext.

This transformation is applied to both the exported content and the preview content in details.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4546

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by ntolia** [comment]
I would expect Exports to match what was captured. Preview is a different case.

**Comment by ryanfkeepers** [comment]
> I would expect Exports to match what was captured. Preview is a different case.

Hmm, that's a good topic for discussion.  We waffle back and forth between whether exports should be exact or human readable.  For example: we're already slicing out most of the stored information in a standard export to create a more digestable experience (there's a flag for providing raw output on top).  Are we better off here going for full human consumption, or partial?

**Comment by ntolia** [comment]
> > I would expect Exports to match what was captured. Preview is a different case.
> 
> Hmm, that's a good topic for discussion. We waffle back and forth between whether exports should be exact or human readable. For example: we're already slicing out most of the stored information in a standard export to create a more digestable experience (there's a flag for providing raw output on top). Are we better off here going for full human consumption, or partial?

Folks might pull these messages into other third-party systems. I think export should try to be as true to the original as possible/feasible for broader use. In particular, stripping out HTML, etc. is definitely not what someone would want but this applies to other components of the conversation too.

**Comment by gmatev** [comment]
> > > I would expect Exports to match what was captured. Preview is a different case.
> > 
> > 
> > Hmm, that's a good topic for discussion. We waffle back and forth between whether exports should be exact or human readable. For example: we're already slicing out most of the stored information in a standard export to create a more digestable experience (there's a flag for providing raw output on top). Are we better off here going for full human consumption, or partial?
> 
> Folks might pull these messages into other third-party systems. I think export should try to be as true to the original as possible/feasible for broader use. In particular, stripping out HTML, etc. is definitely not what someone would want but this applies to other components of the conversation too.

@ryanfkeepers I think you are reading this too literally. It is not about no changes to the payload of the graph call. It is more about the `content` and I guess `subject` fields. If there are other fields returned in the graph call that are not that relevant for the content of the message it should be ok to sanitize some of these to simplify the JSON export. 

If you prefer we can discuss specific examples.

**Comment by ryanfkeepers** [comment]
@gmatev @ntolia Thanks for the reviews.  All changes are now in place.

File: src/pkg/services/m365/api/channels_test.go [hunk 1]
```diff
@@ -565,3 +567,121 @@ func (suite *ChannelsAPIUnitSuite) TestChannelMessageInfo() {
...
+		{
+			name:        "with html",
+			content:     "<body>body<div>in the div</div>end</body>",
+			attachments: attachments,
+			expect:      "body\nin the div\nend",
+			expectErr:   assert.NoError,
+		},
+		{
+			name:        "with html and attachments",
+			content:     "<body>body<div>" + attachML("id1") + attachML("id2") + attachML("idX") + "</div>end</body>",
```
Comment (vkamra) [inline src/pkg/services/m365/api/channels_test.go:665]: @ryanfkeepers - not a blocker here but could we add a couple of test cases with "real content" obtained from Graph API?

**Review by vkamra — APPROVED** [review]
LGTM. Approving to unblock merge. Request we do add a couple of test cases with "real content" from the Graph API while this is merging to increase confidence.

**Pull request merged**